client/my-sites/email/titan-setup-mailbox/index.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * External dependencies
+ */
+import page from 'page';
+import React, { useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { useSelector } from 'react-redux';
+import { useTranslate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import DocumentHead from 'calypso/components/data/document-head';
+import EmailHeader from 'calypso/my-sites/email/email-header';
+import {
+	emailManagementPurchaseNewEmailAccount,
+	emailManagementTitanSetupMailbox,
+} from 'calypso/my-sites/email/paths';
+import getCurrentRoute from 'calypso/state/selectors/get-current-route';
+import { getDomainsBySiteId, hasLoadedSiteDomains } from 'calypso/state/sites/domains/selectors';
+import { getSelectedDomain } from 'calypso/lib/domains';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import { hasTitanMailWithUs } from 'calypso/lib/titan';
+import HeaderCake from 'calypso/components/header-cake';
+import Main from 'calypso/components/main';
+import TitanSetupMailboxForm from 'calypso/my-sites/email/titan-setup-mailbox/titan-setup-mailbox-form';
+import PageViewTracker from 'calypso/lib/analytics/page-view-tracker';
+import QuerySiteDomains from 'calypso/components/data/query-site-domains';
+
+const TitanSetupMailbox = ( { selectedDomainName } ) => {
+	const selectedSite = useSelector( getSelectedSite );
+
+	const currentRoute = useSelector( getCurrentRoute );
+
+	const siteId = selectedSite?.ID ?? null;
+
+	const domains = useSelector( ( state ) => getDomainsBySiteId( state, siteId ) );
+
+	const selectedDomain = getSelectedDomain( { domains, selectedDomainName } );
+
+	const hasTitanSubscription = hasTitanMailWithUs( selectedDomain );
+
+	const areSiteDomainsLoaded = useSelector( ( state ) => hasLoadedSiteDomains( state, siteId ) );
+
+	const analyticsPath = emailManagementTitanSetupMailbox( ':site', ':domain', currentRoute );
+
+	const siteSlug = selectedSite?.slug ?? null;
+
+	const goToCustomerHome = useCallback( () => {
+		page.redirect( `/home/${ siteSlug }` );
+	}, [ siteSlug ] );
+
+	const translate = useTranslate();
+
+	if ( areSiteDomainsLoaded && ! hasTitanSubscription ) {
+		page( emailManagementPurchaseNewEmailAccount( siteSlug, selectedDomainName, currentRoute ) );
```

Comment:
We should probably redirect to `emailManagement()` if the user has a Google subscription. But we can do that in a follow-up.